File: abc/abc160/abc160_c.cs
// https://atcoder.jp/contests/abc160/submissions/11315119

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var input = Console.ReadLine().Split().Select(long.Parse).ToArray();
            var N = input[0];
            var K = input[1];
            var A = Console.ReadLine().Split().Select(long.Parse).ToArray();

            var distance = new long[N];
            for (var i = 0; i < K - 1; i++)
            {
                distance[i] = A[i + 1] - A[i];
            }
            distance[K - 1] = A[0] + N - A[K - 1];

            Console.WriteLine(distance.Sum() - distance.Max());
        }
    }
}

File: abc/abc160/abc160_c.fs
// https://atcoder.jp/contests/abc160/submissions/11749539

let [| N; K |] = stdin.ReadLine().Split() |> Array.map int
let A = stdin.ReadLine().Split() |> Array.map int

let d =
    A
    |> Seq.pairwise
    |> Seq.map (fun (x, y) -> y - x)
    |> Seq.append [ A.[0] + N - A.[K - 1] ]

(d |> Seq.sum) - (d |> Seq.max) |> printfn "%d"
